### DISCUSSION TITLE
What are people working on (April to June 2023)?

### DISCUSSION THREAD

**Discussion by bjchambers** [kaskada-ai/kaskada#272] [General]
It would be nice to know what is being worked on within Kaskada. Rather than trying to assemble a complete roadmap, I thought it could be fun for each contributor to say hi and a brief description of what they're working on for the next few months.

I'll start!

---

I'm Ben Chambers. I tend to focus mostly on the compute engine and query language. In the next few months, I'd like to introduce logical and physical plans with the goal of supporting partitioned execution. I'd also like to cleanup some of the boundaries within the compute engine to make it more extensible and re-usable.

**Comment by therapon** [comment]
I am Theo (Therapon) Skoteiniotis. For the next 3 months I'd like to work on

1. the definition and implementation of Logical and Physical Plans with the goal to then try out 
  *  optimizations (rewrites) at the logical plan level
  *  execution on other systems (in whole or in part)  
2. improvements to realtime execution around long running materializations and dealing with late data 


I'd also like to start experimenting with defining and applying bag semantics on Fenl.

**Comment by epinzur** [comment]
Hello, I'm Eric Pinzur, and my primary focus is to simplify the use and installation of Kaskada. My upcoming priorities for the next few months are as follows:

* Incorporating Azure and Google object storage support into our system (we already support S3)
* Improving our documentation and optimizing the first-time user experience
* Exploring strategies for deploying the service in production environments.

**Comment by jordanrfrazier** [comment]
I'm Jordan Frazier, and mainly focusing on improving our compute engine and query language! In the next few months, I plan to focus on improving our real-time streaming story to enable users to be able to use our service in their production environments. Some of the specifics involve allowing users to directly hook the compute engine directly to their existing Pulsar and Kafka streams (and more, hopefully!), where the engine will then automatically materialize query results to the destination of your choice.
